### ISSUE TITLE
Adding  PDF File as attachment to the Email

### ISSUE THREAD

**Issue by chequille** [einpraegsam/powermail#164]
Hi All reading this,
first I want to say, that this is not a bug, it is just a question if maybe somebody could give a hint how to do this.
I am using powermail and bvt_powermailpdf and this is working together, meaning the mail is sent and bvt_powermailpdf is creating a filled pdf and is storing it to the upload/tx_powermail dorectory.

So far, so good.

My question is whether somebody knows how I can attach this created PDF file to the email?
I tried to use in the typoscript setup:
`plugin.tx_powermail.settings.setup.sender.addAttachment = TEXT
    plugin.tx_powermail.settings.setup.sender.addAttachment {
      value = fileadmin/2example.pdf
                                  
  }`

and tried to change the value within the pdfgen.php to the created pdf filename using

`        $powermailSettings = $GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_powermail.']['settings.'];
        $powermailFilePath = $powermailSettings['setup.']['sender.']['addAttachment.']['value'] = $filename; ,`

but this does not work.

bvt_powermailpdf is using the signal createActionAfterMailDbSaved and i changed this to createActionBeforeRenderView. Unfortunately, none of these did work.
It seems, that the typoscript value cannot be changed or it must be done at another point of the processing chain.

Somebody an idea?
BR
Jürgen

**Comment by einpraegsam** [comment]
Hi Jürgen,

If we are a bit creative, it should be possible to get this to work :)
But first of all, I don't know the pdf-creation extension. So I'm not sure when the PDF is generated and I think the PDF must be created before attaching it to an email.

Nevertheless it should be possible to use a signal sendTemplateEmailBeforeSend in Classes/Domain/Service/SendMailService.php:170 to manipulate $message and add an attachment with `$message->attach(\Swift_Attachment::fromPath('/var/www/html/fileadmin/test.pdf'));`

See https://github.com/einpraegsam/powermailextended/blob/master/ext_localconf.php#L20 for an example how to use this signal.

Hope that helps. If not, please play a bit. I'm sure you will find a solution.

**Comment by chequille** [comment]
Hi Alex,
first thanks for this.
But I tried this and I found, that when using the Signal "sendTemplateEmailBeforeSend", the email is not send anymore.
This is also happening when I use your powermailextended extension.
When I remove the line
$this->signalDispatch(__CLASS__, 'sendTemplateEmailBeforeSend', $signalArguments);
in SendMailService.php, than the email is send again,

It seems, that there is maybe something wrong with the parameter $signalArguments.

Test your own extension. It is not working with powermail 3.22.1 and Typo3 7.6.21

Maybe you have an idea whats wrong.

BR
Jürgen
